odoo/addons/base/transpiler/js_transpiler.py

```diff
@@ -0,0 +1,594 @@
+    """
+    def repl(matchobj):
+        path = matchobj["path"]
+        new_path = path[: path.rfind("/index")] + path[0]
+        return f"require({new_path})"
+    return URL_INDEX_RE.sub(repl, content)
+
+
+def relative_path_to_module_path(url, path_rel):
+    """
+    Convert the relative path into a module path, which is more generic and fancy.
+
+    :param path_rel: a relative path to the current url.
+    :return: module path (@module/...)
+    """
+    url_split = url.split("/")
+    path_rel_split = path_rel.split("/")
+    nb_back = len([v for v in path_rel_split if v == ".."]) + 1
+    result = "/".join(url_split[:-nb_back] + [v for v in path_rel_split if not v in ["..", "."]])
+    return url_to_module_path(result)
+
+
+ODOO_MODULE_RE = re.compile(r"""
+    \/(\*|\/).*\s+              # // or /*
+    @odoo-module\s+             # @odoo-module
+    (alias=(?P<alias>\S+))?\s+  # alias=web.AbstractAction (optional)
+    (default=(?P<default>\S+))? # default=False (optional)
+    """, re.VERBOSE)
+
+
+def is_odoo_module(content):
+    """
+    Detect if the file need to be transpiled.
+    We look for a comment containing @odoo-module.
+
+    :param content: source code
+    :return: is this a odoo module that need transpilation ?
+    """
+    result = ODOO_MODULE_RE.match(content)
+    return bool(result)
+
+
+def get_aliased_odoo_define_content(module_path, content):
+    """
+    To allow smooth transition between the new system and the legacy one, we have the possibility to
+    defined an alternative module name (an alias) that will act as proxy between legacy require calls and
+    new modules.
+
+    Example:
+    If we have a require call somewhere in the odoo source base being:
+    > vat AbstractAction require("web.AbstractAction")
+    we have a problem when we will have converted to module to ES6: its new name will be more like
+    "web/chrome/abstract_action". So the require would fail !
+    So we add a second small modules, an alias, as such:
+    > odoo.define("web/chrome/abstract_action", function(require) {
+    >  return require('web.AbstractAction')[Symbol.for("default")];
+    > });
+
+    To generate this, change your comment on the top of the file.
+
+    .. code-block:: javascript
+
+        // before
+        /** @odoo-module */
+        // after
+        /** @odoo-module alias=web.AbstractAction */
+
+    Notice that often, the legacy system acted like they it did defaukt imports. That's why we have the
+    "[Symbol.for("default")];" bit. If your use case does not need this default import, just do:
+
+    .. code-block:: javascript
+
+        // before
+        /** @odoo-module */
+        // after
+        /** @odoo-module alias=web.AbstractAction default=false */
+
+    :return: the alias content to append to the source code.
+    """
+    matchobj = ODOO_MODULE_RE.match(content)
+    if matchobj:
+        m = matchobj.groupdict()
+        alias = m.get('alias')
```

Comment:
This is overkill, you can index `matchobj`:
```python
alias = matchobj['alias']
```
The result is a string if the group is present, and `None` otherwise.